### DISCUSSION TITLE
VM network problem

### DISCUSSION THREAD

**Discussion by macnack** [utmapp/UTM#3768] [UTM Help (macOS)] [answered]
**Describe the issue**
Network initialization failed. My VM Ubuntu 20.04.4 LTS doesn't have network connection. If I ping 8.8.8.8, I get 'Network is unreachable'. 

**Configuration**
* UTM Version: 3.1.5 (53)
* OS Version: 12.2.1
* Intel or Apple Silicon?  Apple Silicon

**Comment by osy** [comment, marked as answer]
https://mac.getutm.app/gallery/ubuntu-20-04 Check out the troubleshooting section

**Reply by jamesmortensen** [reply]
The troubleshooting section is specific to Ubuntu, but it was helpful in resolving the networking issue I had in a different distribution. For Debian 11, networking died when upgrading from UTM 2.4.1 to UTM 3.1.5.  Based on the troubleshooting information, I observed the network interface had been renamed from enp0s6 to enp0s7.  I edited /etc/network/interfaces and replaced all instances of enp0s6 with enp0s7 and rebooted.  This resolved the networking issue.  Here is what that file looks like so it can help others:

```
# This file describes the network interfaces available on your system
# and how to activate them. For more information, see interfaces(5).

source /etc/network/interfaces.d/*

# The loopback network interface
auto lo
iface lo inet loopback

# The primary network interface
allow-hotplug enp0s7
iface enp0s7 inet static
address 10.0.2.15
netmask 255.255.255.0
gateway 10.0.2.2
dns-nameservers 8.8.8.8 1.1.1.1
route add default gw 10.0.2.2 enp0s7
/sbin/ifconfig enp0s7 10.0.2.15 netmask 255.255.255.0 up
```

**Reply by bergstenarn** [reply]
I experienced the same problem when powering up Debian 10.4 (LDXE) arm64 in UTM on my MacBook Air m1. I resolved the issue essentially by doing what's suggested above, i.e.

1. Power up the Debian VM.
2. Log in as root.
3. Launch a terminal.
4. Check the second interface displayed when running the following:
`ip addr`
This is what it looked like in my case:
```
...
2: enp0s7: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether af:cc:45:10:85:e6 brd ff:ff:ff:ff:ff:ff
```
5. Update file /etc/network/interfaces by replacing all occurrences of enp0sX with whatever the output of ip addr above showed, i.e. enp0s7 in my case.
6. Reboot the VM.
7. Log in and verify that network connectivity exists, for example by launching a terminal and run:
`ping www.google.com`

@osy,  I suggest the UTM team adds a troubleshooting section to page [https://mac.getutm.app/gallery/debian-10-4-ldxe](https://mac.getutm.app/gallery/debian-10-4-ldxe) and tell that so others don't have to spend hours troubleshooting, like I did before I found this post.

FYI here is my configuration:
UTM Version: 3.1.5 (53)
OS Version: 12.3.1
Intel or Apple Silicon? Apple Silicon

**Reply by jamesmortensen** [reply]
I wonder if it is possible to write a script that can be run at boot time that checks the network interface and adjusts it automatically in these scenarios.  

For instance, here is another time consuming task that I recently discovered is automated. 
 Debian cloud images have software on them that detect when we've used qemu-img to resize a qcow2 disk. It automatically resizes the partitions so we don't have to do it manually with qparted.  The package is called growpart.

Let me know if I should open the "auto adjust network interfaces" as a feature request, if it is something UTM can do...
